src/main/java/ru/itis/services/ItemServiceImpl.java

```diff
@@ -22,25 +23,32 @@
     private UserService userService;
 
     @Override
-    public Item addNewItem(ItemForm itemForm) {
-        if (wishListService.findWishListById(itemForm.getWishListId()).isPresent()) {
-            WishList wishList = wishListService.findWishListById(itemForm.getWishListId()).get();
-            Item newItem = Item.builder()
-                    .name(itemForm.getName())
-                    .price(itemForm.getPrice())
-                    .link(itemForm.getLink())
-                    .wishList(wishList)
-                    .build();
-            wishList.getItems().add(newItem);
-            return itemsRepository.save(newItem);
-        } else throw new NullPointerException("No wishlists found");
+    public Item addNewItem(ItemForm itemForm, Long listId) {
+        WishList wishList = getWishList(listId);
+        Item newItem = Item.builder()
+                .name(itemForm.getName())
+                .price(itemForm.getPrice())
+                .link(itemForm.getLink())
+                .wishList(wishList)
+                .build();
+        wishList.getItems().add(newItem);
+        return itemsRepository.save(newItem);
     }
 
     @Override
-    public void removeByName(String itemName) {
+    public void removeByName(String itemName, Long listId) {
+        WishList wishList = getWishList(listId);
         Optional<Item> itemCandidate = itemsRepository.findByName(itemName);
         if (itemCandidate.isPresent()) {
             itemsRepository.deleteById(itemCandidate.get().getId());
+            wishList.getItems().remove(itemCandidate.get());
         } else throw new IllegalArgumentException("Can not find such item");
```

Comment:
Inconsistent exceptions. In some cases you use NoSuchElementException, in others - IllegalArgumentException.

Reply:
see [line 52](https://github.com/AinaArd/wish-list/pull/1/files#diff-17a8d1d4da581f79549a1ea3c34550f6R52)

Reply:
fixed